### PR TITLE
maryia/WEBREL-323/feat: ts migration of base-store

### PR DESCRIPTION
**Pull request by maryia-deriv**
## Changes:

Please provide a summary of the change.

### Screenshots:

Please provide some screenshots of the change.

### CODE REVIEW

File: packages/trader/src/Stores/base-store.ts [hunk 1]
```diff
@@ -1,7 +1,18 @@
 import { action, intercept, observable, reaction, toJS, when, makeObservable } from 'mobx';
 import { isProduction, isEmptyObject } from '@deriv/shared';
-
+import { TCoreStores } from '@deriv/stores/types';
 import Validator from 'Utils/Validator';
+import { getValidationRules } from './Modules/Trading/Constants/validation-rules';
+
+type TValidationRules = ReturnType<typeof getValidationRules> | Record<string, never>;
```
Comment (henry-deriv) [inline packages/trader/src/Stores/base-store.ts:7]: what's the point of `Record<string, never>` ?

**Review by henry-deriv — APPROVED** [review]
lgtm

File: packages/trader/src/Stores/base-store.ts (hunk 1 above)
Comment (maryia-deriv) [reply packages/trader/src/Stores/base-store.ts:7]: @henry-deriv  it's representing an empty object which is validation_rules's initial value in this store :)

**Review by kate-deriv — APPROVED** [review]

File: packages/stores/types.ts [hunk 2]
```diff
@@ -535,6 +535,11 @@ export type TCoreStores = {
     modules: Record<string, any>;
     notifications: TNotificationStore;
     traders_hub: TTradersHubStore;
+    gtm: Record<string, unknown>;
+    pushwoosh: Record<string, unknown>;
+    contract_replay: Record<string, unknown>;
+    chart_barrier_store: Record<string, unknown>;
+    active_symbols: Record<string, unknown>;
```
Comment (farzin-deriv) [inline packages/stores/types.ts:542]: @maryia-deriv I recommend not doing this, We should either type things properly or if out of the scope of this PR just leave it as it was with the TS error for the next person to fix 🙇🏻

**Pull request merged**